File: client_windows_native/Core/AppState.cs
namespace VoiceTyper.Core;

/// <summary>应用状态枚举，与 macOS Swift 版完全一致</summary>
internal enum AppState
{
    /// <summary>启动中</summary>
    Booting,
    /// <summary>需要配置</summary>
    SetupRequired,
    /// <summary>就绪</summary>
    Idle,
    /// <summary>录音中</summary>
    Recording,
    /// <summary>识别中</summary>
    Recognizing,
    /// <summary>正在输入</summary>
    Inserting,
    /// <summary>错误</summary>
    Error
}

internal static class AppStateExtensions
{
    /// <summary>状态的菜单显示文字</summary>
    public static string MenuTitle(this AppState state) => state switch
    {
        AppState.Booting => "启动中...",
        AppState.SetupRequired => "需要配置",
        AppState.Idle => "就绪",
        AppState.Recording => "录音中...",
        AppState.Recognizing => "识别中...",
        AppState.Inserting => "正在输入...",
        AppState.Error => "错误",
        _ => "未知"
    };
}

File: client-server/client_macos_swift/Sources/VoiceTyperApp/Core/AppState.swift
import Foundation

enum AppState: Equatable {
    case booting
    case setupRequired
    /// 权限齐全但服务端尚未就绪（如模型仍在加载），后台自动轮询重连中。
    case connecting
    case idle
    case recording
    case recognizing
    case inserting
    /// 用户主动暂停听写：热键监听与服务轮询均停止，直到用户从菜单恢复。
    case paused
    case error(String)

    var menuTitle: String {
        switch self {
        case .booting:
            return "启动中"
        case .setupRequired:
            return "需要完成授权与设置"
        case .connecting:
            return "连接服务中…"
        case .idle:
            return "就绪"
        case .recording:
            return "录音中..."
        case .recognizing:
            return "识别中..."
        case .inserting:
            return "输入中..."
        case .paused:
            return "已暂停"
        case .error(let message):
            return "错误: \(message)"
        }
    }

    var statusSymbolName: String {
        switch self {
        case .booting:
            return "hourglass"
        case .setupRequired:
            return "exclamationmark.triangle"
        case .connecting:
            return "arrow.triangle.2.circlepath"
        case .idle:
            return "mic"
        case .recording:
            return "mic.fill"
        case .recognizing:
            return "waveform"
        case .inserting:
            return "character.cursor.ibeam"
        case .paused:
            return "mic.slash"
        case .error:
            return "exclamationmark.circle"
        }
    }
}
